src/models/state/mod.rs

```diff
@@ -1566,7 +1568,7 @@ impl GlobalState {
         };
 
         let tip_height = tip.header().height;
-        if tip_height < MIN_BLOCK_HEIGHT_FOR_SYNCING.into() {
+        if tip_height < (SYNC_CHALLENGE_POW_WITNESS_LENGTH as u64).into() {
```

Comment:
Since `SYNC_CHALLENGE_POW_WITNESS_LENGTH` and `SYNC_CHALLENGE_NUM_BLOCK_PAIRS` are independent constants, and since sync mode only makes sense if the claimed tip height is larger than both, shouldn't we check `tip_height` against the minimum of these constants?

```suggestion
        if tip_height < (usize::min(SYNC_CHALLENGE_POW_WITNESS_LENGTH,SYNC_CHALLENGE_NUM_BLOCK_PAIRS) as u64).into() {
```